src/main/java/uk/gov/justice/digital/delius/transformers/AssessmentTransformer.java

```diff
@@ -10,33 +10,34 @@
 
 public class AssessmentTransformer {
 
-    public static OffenderAssessments assessmentsOf(Offender offender, Optional<OGRSAssessment> OGRSAssessment, Optional<OASYSAssessment> OASYSAssessment) {
+    public static OffenderAssessments assessmentsOf(Offender offender, Optional<OGRSAssessment> ogrsAssessment, Optional<OASYSAssessment> OASYSAssessment) {
         return OffenderAssessments
             .builder()
             .rsrScore(offender.getDynamicRsrScore())
-            .ogrsScore(getOGRSScore(OGRSAssessment, OASYSAssessment))
+            .ogrsScore(getOGRSScore(ogrsAssessment, OASYSAssessment))
+            .orgsLastUpdate(ogrsAssessment.map(OGRSAssessment::getLastUpdatedDate).orElse(null))
             .build();
     }
 
-    private static Integer getOGRSScore(Optional<OGRSAssessment> OGRSAssessment, Optional<OASYSAssessment> OASYSAssessment) {
+    private static Integer getOGRSScore(Optional<OGRSAssessment> oGRSAssessment, Optional<OASYSAssessment> oASYSAssessment) {
 
-        Integer OASYSAssessmentScore = OASYSAssessment.map(o -> o.getOGRSScore2()).orElse(null);
-        LocalDate OASYSAssessmentDate = OASYSAssessment.map(o -> o.getAssessmentDate()).orElse(null);;
+        Integer oASYSAssessmentScore = oASYSAssessment.map(OASYSAssessment::getOGRSScore2).orElse(null);
+        LocalDate oASYSAssessmentDate = oASYSAssessment.map(OASYSAssessment::getAssessmentDate).orElse(null);
 
-        return OGRSAssessment.map(OGRS -> {
-            final var OGRSAssessmentScore = OGRS.getOGRS3Score2();
-            if (null == OASYSAssessmentScore) {
-                return OGRSAssessmentScore;
-            }
 
-            if (null != OGRSAssessmentScore) {
-                final var OGRSAssessmentDate = OGRS.getAssessmentDate();
-                if (OGRSAssessmentDate.isAfter(OASYSAssessmentDate)) {
-                    return OGRSAssessmentScore;
+        return oGRSAssessment.map(OGRS -> {
+            var ogrsassessmentscore = OGRS.getOGRS3Score2();
+
+            if (null == oASYSAssessmentScore) {
+                return ogrsassessmentscore;
+            }
+            if (null != ogrsassessmentscore) {
+                if (OGRS.getAssessmentDate().isAfter(oASYSAssessmentDate)) {
+                    return ogrsassessmentscore;
                 }
             }
             return null;
-        }).orElse(OASYSAssessmentScore);
+        }).orElse(oASYSAssessmentScore);
```

Comment:
See what you mean now, this logic is a bit confusing - maybe worth adding a comment to say what it's doing?

My understanding is that it returns the score from either the OGRS assessment or the OASys assessment based on the following rules:
* If only an OGRS assessment is present, return that.  
* If both assessments are present, return the one with the most recent assessment date.  
* Otherwise return the score from the OASys assessment.

Reply:
I did not change this logic . I have just  adjusted the  variable names to camelCase